File: src/main/java/pl/tsimur/jpatest/service/OrganizationService.java
package pl.tsimur.jpatest.service;

import pl.tsimur.jpatest.model.Organization;

import java.util.List;

public interface OrganizationService {

    Integer save(Organization org);

    Organization find(Integer nip);

    List<Organization> findAllByName(String name);

    List<Organization> findAllByAddress(String address);

    List<String> findAllOrgNamesWithPensioners();

    void updateAddress(Integer nip, String address);

    void delete(Integer nip);
}

File: src/main/kotlin/pl/timur/jpatest/service/OrganizationService.kt
package pl.timur.jpatest.service

import pl.timur.jpatest.model.Organization

interface OrganizationService {
    fun save(org: Organization): Int

    fun find(nip: Int): Organization?

    fun findAllByName(name: String): List<Organization>

    fun findAllByAddress(address: String): List<Organization>

    fun findAllOrgNamesWithPensioners(): List<String>

    fun updateAddress(nip: Int, address: String)

    fun delete(nip: Int)
}